index.ts

```diff
@@ -32,18 +30,44 @@ const defaultOptions = {
 	darkTheme: 'dark',
 }
 
+/**
+ * Retrieves all non-atrule nodes from the provided node and its children.
+ * @param {ChildNode} node - The node to traverse.
+ * @returns {Rule[]} - An array containing non-atrule nodes.
+ */
+const notAtruleNode = (node: ChildNode): Rule[] => {
+	const collectedRules: Rule[] = []
+
+	/**
+	 * Recursively collects non-atrule nodes.
+	 * @param {ChildNode} currentNode - The current node to examine.
+	 */
+	const collectNonAtrules = (currentNode: ChildNode) => {
+		if (currentNode.type === 'atrule') {
```

Comment:
I had to do the following change to make TSC happy:
```suggestion
		if (currentNode.type === 'atrule' && currentNode.nodes) {
```